test-case/test-jack-detection-playback-capture.sh

```diff
@@ -0,0 +1,164 @@
+# * Repeat for both headphone and headset jacks.
+# * Repeat for both HDMI and DisplayPort if available.
+
+## Case Steps
+# 1. Ensure the USB relay switch is configured to control the jack detection header.
+# 2. Set the USB relay switch to state off (0), simulate plugging in the headset.
+# 3. Run aplay/arecord command to play/record audio.
+# 4. Check the jack detection status via amixer. The status should indicate **on**.
+# 5. Set the USB relay switch to state on (1), simulate unplugging the headset from the jack.
+# 6. Check the jack detection status via amixer. The status should indicate **off**.
+# 7. Check dmesg for any unexpected errors.
+#
+# Repeat for both headphone and headset jacks.
+# Repeat for all pipelines.
+
+TESTDIR=$(cd "$(dirname "${BASH_SOURCE[0]}")/.." && pwd)
+TESTLIB="${TESTDIR}/case-lib"
+
+# shellcheck source=case-lib/lib.sh
+source "${TESTLIB}/lib.sh"
+
+OPT_NAME['t']='tplg'     OPT_DESC['t']="tplg file, default value is env TPLG: $TPLG"
+OPT_HAS_ARG['t']=1         OPT_VAL['t']="$TPLG"
+
+OPT_NAME['r']='round'     OPT_DESC['r']='round count'
+OPT_HAS_ARG['r']=1         OPT_VAL['r']=1
+
+OPT_NAME['d']='duration' OPT_DESC['d']='arecord duration in second'
+OPT_HAS_ARG['d']=1         OPT_VAL['d']=2
+
+OPT_NAME['l']='loop'     OPT_DESC['l']='option of speaker-test'
+OPT_HAS_ARG['l']=1         OPT_VAL['l']=3
+
+OPT_NAME['s']='sof-logger'   OPT_DESC['s']="Open sof-logger trace the data will store at $LOG_ROOT"
+OPT_HAS_ARG['s']=0             OPT_VAL['s']=1
+
+OPT_NAME['u']='relay'     OPT_DESC['u']='name of usbrelay switch, default value is HURTM_2'
+OPT_HAS_ARG['u']=1         OPT_VAL['u']="HURTM_2"
+
+func_opt_parse_option "$@"
+
+tplg=${OPT_VAL['t']}
+duration=${OPT_VAL['d']}
+relay=${OPT_VAL['u']}
+tcnt=${OPT_VAL['l']}
+round_cnt=${OPT_VAL['r']}
+
+check_control_switch_state()
+{
+    # Check the state of the switch using amixer.
+    # The switch name is passed as the first argument, and the expected state (on/off)
+    # is passed as the second argument.
+    # Returns 0 if the state matches, 1 otherwise.
+    local switch_name="$1"
+    local expected_switch_state="$2"
+    local switch_state
+
+    switch_state=$(echo -e $(amixer -c 0 contents | grep -i "$switch_name .* *jack" -A 2) | sed -n '1s/^.*values=//p')
+    dlogi "$switch_name switch is: $switch_state"
+
+    if [[ "$expected_switch_state" == "$switch_state" ]]; then
+        return 0
+    else
+        return 1
+    fi
+}
+
+main()
+{
+    func_pipeline_export "$tplg" "type:playback"
+
+    setup_kernel_check_point
+
+    start_test
+
+    logger_disabled || func_lib_start_log_collect
+
+    # check if usbrelay tool is installed
+    command -v usbrelay || {
+        skip_test "usbrelay command not found. Please install usbrelay to control the mic privacy switch."
+    }
+
+    dlogi "Reset - plug jack audio"
+    usbrelay_switch "$relay" 0
+
+    for round in $(seq 1 "$round_cnt")
+    do
+        for idx in $(seq 0 $((PIPELINE_COUNT - 1)))
+        do
```

Comment:
It would be more readable to move this to some new `test_one_idx()`  function (better name wanted), that would avoid a lot of indentation and a lot of distance between the start and end of the for loops.